biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -235,6 +298,93 @@ end subroutine LoggingMortality_frac
+      character(len=64), intent(in) :: hlm_harvest_catnames(:) ! names of hlm harvest categories
+      integer, intent(in) :: use_history        ! patch level anthro_disturbance_label
+      real(r8), intent(in) :: secondary_age     ! patch level age_since_anthro_disturbance
+      real(r8), intent(in) :: frac_site_primary
+      real(r8), intent(out) :: harvest_rate
+
+      ! Local Variables
+      integer :: h_index   ! for looping over harvest categories
+      integer :: icode   ! Integer equivalent of the event code (parameter file only allows reals)
+
+      ! Parameters
+      real(r8), parameter :: months_per_year = 12.0
+
+     ! determine the annual hlm harvest rate for the current cohort based on patch info
+     harvest_rate = 0._r8
+     do h_index = 1,hlm_num_lu_harvest_cats
+        if (use_history .eq. primaryforest) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_VH1" .or. &
+                hlm_harvest_catnames(h_index) .eq. "HARVEST_VH2") then
+              harvest_rate = harvest_rate + hlm_harvest(h_index)
+           endif
+        else if (use_history .eq. secondaryforest .and. &
+             secondary_age >= secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH1") then
+              harvest_rate = harvest_rate + hlm_harvest(h_index)
+           endif
+        else if (use_history .eq. secondaryforest .and. &
+             secondary_age < secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH2" .or. &
+                hlm_harvest_catnames(h_index) .eq. "HARVEST_SH3") then
+              harvest_rate = harvest_rate + hlm_harvest(h_index)
+           endif
+        endif
+     end do
+
+     ! normalize by site-level fractio primary or secondary forest fraction
+     ! since harvest_rate is specified in fraction of the gridcell
+     ! also need to put a cap so as not to harvest more primary or secondary area than there is in a gridcell
+     if (use_history .eq. primaryforest) then
+        if (frac_site_primary .gt. fates_tiny) then
+           harvest_rate = min((harvest_rate / frac_site_primary),frac_site_primary)
+        else
+           harvest_rate = 0._r8
+        endif
+     else
+        if ((1._r8-frac_site_primary) .gt. fates_tiny) then
+           harvest_rate = min((harvest_rate / (1._r8-frac_site_primary)),&
+                (1._r8-frac_site_primary))
+        else
+           harvest_rate = 0._r8
+        endif
+     endif
+
+     ! calculate today's harvest rate
+     ! whether to harvest today has already been determined by IsItLoggingTime
+     ! for icode == 2, icode < 0, and icode > 10000 apply the annual rate one time (no calc)
+     ! Bad logging event flag is caught in IsItLoggingTime, so don't check it here
+     icode = int(logging_event_code)
+     if(icode .eq. 1) then
+        ! Logging is turned off - not sure why we need another switch
```

Comment:
indeed. This does cause extra confusion.

Reply:
I think we left this in to allow the old event-based harvesting code to work.  but maybe we need to get rid of that and make the timeseries code the only way to do the harvest?

Reply:
I think both options can be useful, especially since the fates logging parameters are useful for determining harvest date and the other specifics that are not in the hlm harvest inputs.
This particular value can be eliminated, but it is rather inconsequential, unless set accidentally, so it was left in.

Reply:
fair enough!